docs/data/data-grid/editing/editing.md

```diff
@@ -4,266 +4,411 @@ title: Data Grid - Editing
 
-If you are using a `valueGetter` to extract the value from a nested object, then a `valueSetter` also needs to be provided.
-The first one receives the row object and must return the value to be displayed in the cell.
-In the other side, the second one does the inverse, receiving the new value entered and returning the updated row.
+In the following demo, only the rows with an even `Age` value are editable.
+The editable cells have a green background for better visibility.
 
-The following demo shows how these two functions can be used:
+{{"demo": "IsCellEditableGrid.js", "bg": "inline"}}
 
-{{"demo": "ValueGetterSetterGrid.js", "bg": "inline"}}
+## Value parser and value setter
 
-> Calling the `valueSetter` is the last step in the saving process.
-> The [validation](/components/data-grid/editing/#client-side-validation) will still be called with the values before they pass through the setter.
+A value parser allows to modify the value entered by the user.
+It can be used, for instance, to convert the value sent by the edit component to a different format.
+To do so, use the `valueParser` property of the column definition:
 
-### Client-side validation
+```tsx
+const columns: GridColDef[] = [
+  {
+    valueParser: (value: GridCellValue, params: GridCellParams) => {
+      return value.toLowerCase();
+    },
+  },
+];
+```
 
-To validate the value in the cells, first add a `preProcessEditCellProps` callback to the [column definition](/api/data-grid/grid-col-def/) of the field to validate.
-Once it is called, validate the value provided in `params.props.value`.
-Then, return a new object contaning `params.props` and also the `error` attribute set to true or false.
-If the `error` attribute is true, the value will never be committed.
+On the other hand, a value setter customizes how the row is updated with the new value.
+If you are already using a `valueGetter` to extract the value from a nested object, then the `valueSetter` will do the inverse task.
+To do so, use the `valueSetter` property of the column definition.
+It will be called with an object containing the new value to be saved and the row where this value belongs to.
 
 ```tsx
 const columns: GridColDef[] = [
   {
-    field: 'firstName',
-    preProcessEditCellProps: (params: GridPreProcessEditCellProps) => {
-      const hasError = params.props.value.length < 3;
-      return { ...params.props, error: hasError };
+    valueSetter: (params: GridValueSetterParams) => {
+      const [firstName, lastName] = params.value!.toString().split(' ');
+      return { ...params.row, firstName, lastName };
     },
   },
 ];
 ```
 
-Here is an example implementing an email validation:
+In the following demo, both value parser and value setter were defined for the "Full name" column.
+The value parser converts the value entered to uppercase while the value setter splits the value and saves it correctly into the row model:
```

Comment:
I changed "converts the value entered to uppercase" -> "capitalizes" because the demo is not converting everything to uppercase now, only the first letter of each word.